MetaMask/metamask-mobile
# fix: refactor mechanism for sending analytics events

## **Description**

Filter out nested objects like `{ anonymous: true, value: "ETH" }` instead of a single value `ETH` that are showing up in analytics events since the migration from Mixpanel to Segment.
## **Related issues**

Fixes: #9233 

## **Manual testing steps**

1. Go to Send Page
2. Send a transaction
3. Should see data in analytics dashboard for `active_currency` as single value and not an object

## **Screenshots/Recordings**



### **Before**
<img width="1495" alt="OLD" src="https://github.com/MetaMask/metamask-mobile/assets/141057783/310c976c-2a29-47a1-b65e-2da876c63332">



### **After**

<img width="1493" alt="NEW" src="https://github.com/MetaMask/metamask-mobile/assets/141057783/9980b6ac-a0a4-4b0c-ab58-027de7f838cc">


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
